### PR TITLE
fix(core): additional null check when hashing lockfile

### LINKED ISSUE #13776: Cannot read properties of undefined (reading 'data') when migrating from v14.6.0 to v15.3+
### Current Behavior

Upon trying to run `serve` after migrating to version 15.3.x, I get an error when trying to serve a react application. Resetting to v14.6.0 and updating to version as recent as 15.2.4 works correctly. 

### Expected Behavior

Should be able to run react application in development mode as in prior versions.

### Github Repo

private repo

### Steps to Reproduce

1. from a repo with a react app and react, other lib modules running 14.6.0, run `npx nx migrate @nrwl/workspace@15.2.4`
2. run `npx nx migrate --run-migrations` when complete`
3. run `npx nx serve <appname>`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 18.12.1
   OS   : darwin x64
   npm  : 8.19.2
   
   nx : 15.2.4
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.2.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.2.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.2.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.2.4
   @nrwl/js : 15.2.4
   @nrwl/linter : 15.2.4
   @nrwl/nest : Not Found
   @nrwl/next : 15.2.4
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.2.4
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.2.4
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.2.4
   @nrwl/web : 15.2.4
   @nrwl/webpack : 15.2.4
   @nrwl/workspace : 15.2.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### Failure Logs

```shell
>  NX   Cannot read properties of undefined (reading 'data')
```

### Additional Information

I tried stepping forward through releases up until v15.3.0 and the releases prior to this all will serve the react application. there are no lines in my app that refer to a `data` attribute on an object, unguarded.

### LINKED ISSUE #13777: nx fails after angular:cypress-component-configuration, when using npm
### Current Behavior
after `npx nx generate @nrwl/angular:cypress-component-configuration --project=compotest --generateTests` most nx commands start creating the following output:
``` 
 >  NX   Cannot read properties of undefined (reading 'data')

   
   Because of the error the Nx daemon process has exited. The next Nx command is going to restart the daemon process.
   If the error persists, please run "nx reset".
```

Running `nx reset`does not change anything.

It will go away after running `yarn install` or removing `@cypress/webpack-dev-server` and running `npm i`.

### Expected Behavior

run all nx commands as usual, without needing to have a yarn.lock file

### Github Repo

https://github.com/xel-x/nx-angular-15_3_3

### Steps to Reproduce

Repository was created using the following command, so instead of cloing it, everyone can create it on their own:

```
npx create-nx-workspace@15.3.3 compotest2 --preset=angular-monorepo --appName=compotest --interactive=false --style=css --nx-cloud=false
```
1. npm i
2. npx nx generate @nrwl/angular:cypress-component-configuration --project=compotest --generateTests
3. run any nx command. like `npx nx run compotest:component-test`or `npx nx report`

### Nx Report

```shell
before using yarn: 

>  NX   Cannot read properties of undefined (reading 'data')

   
   Because of the error the Nx daemon process has exited. The next Nx command is going to restart the daemon process.
   If the error persists, please run "nx reset".

after using yarn install:

   Node : 16.16.0
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 15.3.3
   @nrwl/angular : 15.3.3
   @nrwl/cypress : 15.3.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.3.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.3.3
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.3.3
   @nrwl/js : 15.3.3
   @nrwl/linter : 15.3.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 15.3.3
   @nrwl/workspace : 15.3.3
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### Failure Logs

_No response_

### Additional Information

The issue is instantly resolved when using yarn install.
It will appear again as soon as yarn.lock is deleted.

### PR DESCRIPTION
**Pull request by AgentEnder**
## Related Issue(s)

Fixes #13776
Fixes #13777

### CODE REVIEW

File: packages/nx/src/lock-file/utils/hashing.ts [hunk 1]
```diff
@@ -34,29 +34,37 @@ function hashExternalNode(node: ProjectGraphExternalNode, graph: ProjectGraph) {
...
-
     // collect all dependencies' hashes
-    traverseExternalNodesDependencies(node.name, graph, hashingInput);
-    node.data.hash = defaultHashing.hashArray(hashingInput.sort());
+    const externalDependencies = traverseExternalNodesDependencies(
+      node.name,
+      graph,
+      new Set([hashKey])
+    );
+    node.data.hash = defaultHashing.hashArray(Array.from(externalDependencies));
```
Comment (FrozenPandaz) [inline packages/nx/src/lock-file/utils/hashing.ts:43]: I believe the order is important so that it's consistent? Can we keep the sorting?

**Review by FrozenPandaz — DISMISSED** [review]

File: packages/nx/src/lock-file/utils/hashing.ts (hunk 1 above)
Comment (AgentEnder) [reply packages/nx/src/lock-file/utils/hashing.ts:43]: Thanks! I didn't even see the `.sort()` 😦

**Review by meeroslav — APPROVED** [review]
Looks good to me. Thank you!

**Comment by crusoexia** [comment]
When would this get release?

**Pull request merged**